### PR TITLE
Remove old Runner

### LINKED ISSUE #5003: Fully remove (Git)Runner
This tracks the remaining issues before this can happen:

- [x] #4996 
- [x] #5002 
- [x] `customremotes` code depends on them #5142 
- [x] `datalad/support/archive_utils_patool.py` injects a Runner-based execution helper into `patool` #5141 
- [x] https://github.com/datalad/datalad-container/pull/141
- [ ] `datalad/tests/test_cmd.py` can be removed in full, when the classes are gone
- [ ] `datalad/tests/test_protocols.py` 
- [ ] large parts of `datalad.cmd` contain the runner implementation and various helpers that can all be deleted
- [ ] "protocols" will no longer be a thing and can be stripped from
    -  `datalad/customremotes/tests/__init__.py`
    -  `datalad/customremotes/base.py` 
    -  `datalad/customremotes/tests/test_archives.py`
    -  `datalad.interface/common_cfg.py`
    -  `datalad/support/gitrepo.py`
    -  `datalad/support/{protocol.py,tests/test_protocol.py}`, (delete in full)
    -  `datalad/support/tests/test_gitrepo.py`
- [ ] `GitRepo._run_git_custom_command` must be removed
- [x] Fix up command execution helpers in AnnexRepo (this could be a bottomless pit), in particular `_run_annex_command_json()` relies on Runner logging features. #5163

### PR DESCRIPTION
**Pull request by mih**
Fixed #5003

### CODE REVIEW

**Review by kyleam — APPROVED** [review]
Thanks, I'm very happy to see this.  All the commits look good to me (with a boundary fixup pushed for "BF: Rectify test assumption").

> Subject: [PATCH 2/6] RF: Remove run_gitcommand_on_file_list_chunks()
>
> Made obsolete by GitWitlessRunner.run_on_file_list_chunks()

Dropping it right away, rather than leaving it around with a DeprecationWarning, is fine with me.  It seems unlikely to be used by outside callers, and a quick search across known extensions and on github doesn't bring up anything.

**Comment by kyleam** [comment]
Oy, I just now clicked through to the crawler failure.  It has a good number of import errors, which I think are all coming from this part of `datalad_crawler/base.py`:

```python
from datalad.cmd import Runner
from datalad.support.protocol import DryRunProtocol


def get_runner(*args, **kwargs):
    if cfg.obtain('datalad.crawl.dryrun', default=False):
        kwargs = kwargs.copy()
        kwargs['protocol'] = DryRunProtocol()
    return Runner(*args, **kwargs)
```

https://github.com/datalad/datalad/pull/5229/checks?check_run_id=1538522651

File: datalad/customremotes/base.py [hunk 1]
```diff
@@ -66,131 +63,6 @@ def get_function_nargs(f):
         return len(argspec.args) - 1
 
 
-class AnnexExchangeProtocol(ProtocolInterface):
```
Comment (yarikoptic) [inline datalad/customremotes/base.py:69]: bye bye my Dear friend, you were a good help!  I can only hope that I will not miss you, in log and Joey we trust!

**Comment by mih** [comment]
> Oy, I just now clicked through to the crawler failure. It has a good number of import errors, which I think are all coming from this part of `datalad_crawler/base.py`:

I started a PR to perform the necessary RF: https://github.com/datalad/datalad-crawler/pull/89

**Comment by mih** [comment]
@yarikoptic has merged the crawler fix. thx! All good now!

**Comment by mih** [comment]
OK, this is it!

I hope the fall-out over the coming days is not to intense, but I think it is good to merge this now and give it sufficient time to let the dust settle before a 0.14.

Thanks for all the bits you contributed over the years :eyes: to make this possible. Witless we will be from now .

**Pull request merged**